types/instance.go

```diff
@@ -70,3 +80,75 @@ func (i Instance) GetItem() interface{} {
 func (i Instance) GetType() string {
 	return "instance"
 }
+
+type JsonResponseBody struct {
+	Json map[string]interface{}
+}
+
+type RemoteResponse struct {
+	Code  int
+	Body  string
+	Json  JsonResponseBody
+	Error bool
+}
+
+// GetUrl returns the result of an HTTP request to the instance
+func (i Instance) GetUrl(path string, port string) RemoteResponse {
+	var uri string
+	checkPort := "80"
+
+	if port == "" {
+		uri = fmt.Sprintf("http://%s%s", i.IpAddress, path)
+	} else {
+		uri = fmt.Sprintf("http://%s:%s%s", i.IpAddress, port, path)
+		checkPort = port
+	}
+
+	if tcp_port_test(i.IpAddress, checkPort) == false {
+		return RemoteResponse{999, "", JsonResponseBody{}, true}
```

Comment:
For me `-1` is more like an error code than `999`. What do you think does it make sense to use it? Or `0` is  not a valid response code also so if `Code` has not set means the `RemoteResponse` was not a remote call.

Reply:
The reason I went with 999 is due to how HTTP codes are already defined.  The integer ranges of 200-399 is success and 400-599 are errors. So, it's simpler to do `X > 400` than to do `X == -1 && X > 400`.

I don't mind using 0 or -1 as error values. I chose a different value just to simplify the conditional logic a bit.

Now that I've explained my reasoning, please let me know if you'd still like me to rework this. 🙂

Reply:
Thanks, that was my thought :) For me `0` is the default value which means no value in this case. To understand `999` consumers have to read source code. But maybe @keyki has some input here :)